internal/parsers/table/csv/calls.go

```diff
@@ -79,19 +83,42 @@ func (t *Table) SubDivName(countryCode *locodedb.CountryCode, code string) (stri
 	return rec.name, nil
 }
 
+func utf8encoding(subdivCountry string) (string, error) {
+	res, err := charmap.Windows1256.NewDecoder().String(subdivCountry)
+	if err == nil && !strings.Contains(res, "\uFFFD") && !strings.Contains(res, "\u0000") && !strings.Contains(res, "?") {
+		return res, nil
+	}
+
+	res, err = charmap.ISO8859_1.NewDecoder().String(subdivCountry)
+	if err == nil && !strings.Contains(res, "\uFFFD") && !strings.Contains(res, "\u0000") && !strings.Contains(res, "?") {
+		return res, nil
+	}
+
+	return subdivCountry, errInvalidSubName
+}
+
 func (t *Table) initSubDiv() (err error) {
 	t.subDivOnce.Do(func() {
 		t.mSubDiv = make(map[subDivKey]subDivRecord)
 
 		err = t.scanWords([]string{t.subDivPath}, subDivFldNum, func(words []string) error {
+			subdiv := words[subDivName]
+			if !utf8.ValidString(subdiv) {
+				subdiv, err = utf8encoding(subdiv)
+			}
+
 			t.mSubDiv[subDivKey{
 				countryCode: words[subDivCountry],
 				subDivCode:  words[subDivSubdivision],
 			}] = subDivRecord{
```

Comment:
But you still add it here. While the name is incorrect and we know about it. Error should be handled immediately, I think.